x/ccv/provider/keeper/msg_server.go

```diff
@@ -309,16 +293,55 @@ func (k msgServer) SetConsumerCommissionRate(goCtx context.Context, msg *types.M
 	return &types.MsgSetConsumerCommissionRateResponse{}, nil
 }
 
-func (k msgServer) ConsumerModification(goCtx context.Context, msg *types.MsgConsumerModification) (*types.MsgConsumerModificationResponse, error) {
-	if k.GetAuthority() != msg.Authority {
-		return nil, errorsmod.Wrapf(types.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Authority)
+// RegisterConsumer registers a consumer chain
+func (k msgServer) RegisterConsumer(goCtx context.Context, msg *types.MsgRegisterConsumer) (*types.MsgRegisterConsumerResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	consumerId := strconv.FormatUint(k.Keeper.FetchAndIncrementConsumerId(ctx), 10)
+
+	k.Keeper.SetConsumerIdToRegistrationRecord(ctx, consumerId, *msg.RegistrationRecord)
+	k.Keeper.SetConsumerIdToOwnerAddress(ctx, consumerId, msg.Signer)
+	k.Keeper.SetConsumerIdToPhase(ctx, consumerId, Registered)
+
+	return &types.MsgRegisterConsumerResponse{ConsumerId: consumerId}, nil
+}
+
+// InitializeConsumer initializes a consumer chain
+func (k msgServer) InitializeConsumer(goCtx context.Context, msg *types.MsgInitializeConsumer) (*types.MsgInitializeConsumerResponse, error) {
```

Comment:
Is it possible to use this message to update only parts of the InitializationRecord? For example, send a message with all the fields in InitializationRecord unset but for the spawnTime. The result is just updating the spawnTime and leaving the other fields unchanged. 
Same for `UpdateConsumer`

Reply:
This is possible. Seems safer if they have to set **everything** so they do not by accident pick up an old parameter that they did not want to use. But, I've no strong feelings on this. Let me know if you've a preference.

Reply:
My view is that users don't need to query the chain to get the values and then to avoid race condition. If they just one to update one field, they should be able to do so without needing the know the other fields.